addons/io_hubs_addon/components/hubs_component.py

```diff
@@ -21,6 +23,11 @@ class HubsComponent(PropertyGroup):
         'icon': 'icon.png'
     }
 
+    # Properties defined here are for internal use and won't be displayed by default in components or exported.
+
+    # Version of the add-on this component was created with.
+    addon_version: IntVectorProperty(size=3)
```

Comment:
The way I see this is `HubsComponent` having another definition for a version and a class method to get it, same as we are doing for the other definitions:

```python
    _definition = {
        # The name that will be used in the GLTF file MOZ_hubs_components object when exporting the component.
        'name': 'template',
        # Name to be used in the panels, if not set the component name will be used
        'display_name': 'Hubs Component Template',
        # Category that is shown in the "Add Component" menu
        'category': Category.MISC,
        # Node type to where the component will be registered
        'node_type': NodeType.NODE,
        # Panel types where this component will be shown
        'panel_type': [PanelType.OBJECT],
        # The dependencies of this component (by id). They will be added as a result of adding this component.
        'deps': [],
        # Name of the icon to load. It can be a image file in the icons directory or one of the Blender builtin icons id
        'icon': 'icon.png',
        # Version of the component. This will be used to trigger component migrations.
        'version': (0, 0, 1)
    }

    @classmethod
    def get_version(cls):
        return cls.__get_definition('version', (0, 0, 0))
```

and an instance property `version`:
```python
version: IntVectorProperty(size=3)
```

then in `handlers` I think we could get rid of the GLOBAL/LOCAL (not sure though as we are using it for the report in the migrate methods and that might be local specific?) parameter and just compare the class version against the component version and migrate the component whenever they differ:
```python
component.version < component.__class__.get_version()
```

and we can probably refactor some code there as most of the migration cases will look something like where only the migration host changes:
```python
    component_version = component.__class__.get_version()
    try:
        if version < component_version:
            migration_occurred |= True
            component.migrate(version, scene, migration_report)
            component.version = component_version
```
I think you mentioned that there is some case when we still need to compare against the global add-on version but I can't remember when was that.